build-system/babel-plugins/babel-plugin-amp-mode-transformer/index.js

```diff
@@ -24,15 +24,25 @@ const {dirname, join, relative, resolve} = require('path').posix;
 // This plugin is only executed when bundling for production builds (minified).
 module.exports = function ({types: t}) {
   let getModeFound = false;
+  let modeNamespaceFound = false;
+
   return {
     visitor: {
       ImportDeclaration({node}, state) {
         const {source, specifiers} = node;
         if (!source.value.endsWith('/mode')) {
           return;
         }
+
+        if (source.value === '#core/mode') {
```

Comment:
Would the plugin currently handle if say `#core/foobar` had an `import * as mode from './mode';`?

Reply:
nope :/

Reply:
Could we solve that?

Reply:
sure, we could resolve the filename and then check against `src/core/mode.js`, but given that we plan to always use `import * as mode from '#core/mode'` I thought it wouldn't be that important

Reply:
If it's not much more complexity to do so, I feel like it could be worthwhile (there are also files in `src/` still that import from core via relative path), but the marginal gain is pretty tiny so if it's more than a few minutes to tweak I say 🚢 it

Reply:
This PR is blocked on https://github.com/ampproject/amphtml/pull/35477

Reply:
SGTM.
Doing a find-replace on `%s/from '([^']*/core|\.)/mode';/from '#core/mode';` would probably cover the remaining cases